datafusion/physical-plan/src/aggregates/row_hash.rs

```diff
@@ -128,68 +129,32 @@ struct SkipAggregationProbe {
     // ========================================================================
     /// Number of processed input rows (updated during probing)
     input_rows: usize,
-    /// Number of total group values for `input_rows` (updated during probing)
-    num_groups: usize,
 
-    /// Flag indicating further data aggregation may be skipped (decision made
-    /// when probing complete)
-    should_skip: bool,
-    /// Flag indicating further updates of `SkipAggregationProbe` state won't
-    /// make any effect (set either while probing or on probing completion)
-    is_locked: bool,
-
-    /// Number of rows where state was output without aggregation.
-    ///
-    /// * If 0, all input rows were aggregated (should_skip was always false)
-    ///
-    /// * if greater than zero, the number of rows which were output directly
-    ///   without aggregation
-    skipped_aggregation_rows: metrics::Count,
+    /// Number of unique hash, which represents the cardinality of the group values
+    unique_hashes_count: HashSet<u64>,
```

Comment:
What's the main case when the `group_values.len()` is not enough and actual hashes needed?

And if they are really required, this HashSet should be accounted in operators memory reservation.

Reply:
> What's the main case when the group_values.len() is not enough and actual hashes needed?

Even if the existing group is low, we still need to accumulate previous hashes, so we can calculate later on when the group is high enough.

`unique_hashes_count.len()` is the role similar to `group_values.len()`